src/serve.rs

```diff
@@ -65,18 +89,918 @@ async fn table(
     };
 
     tracing::info!("URL: {}", url);
-    let select = sql::parse(&url);
+    let select = parse(&url)?;
     tracing::info!("select {:?}", select);
+
     match get::get_rows(&state.config, &select, "page", &format).await {
         Ok(x) => match format {
-            "html" => Html(x).into_response(),
-            "json" => ([("content-type", "application/json; charset=utf-8")], x).into_response(),
-            "pretty.json" => x.into_response(),
+            "html" => Ok(Html(x).into_response()),
+            "json" => {
+                Ok(([("content-type", "application/json; charset=utf-8")], x).into_response())
+            }
+            "pretty.json" => Ok(x.into_response()),
             _ => unreachable!("Unsupported format"),
         },
         Err(x) => {
             tracing::info!("Get Error: {:?}", x);
-            (StatusCode::NOT_FOUND, Html("404 Not Found".to_string())).into_response()
+            Ok((StatusCode::NOT_FOUND, Html("404 Not Found".to_string())).into_response())
+        }
+    }
+}
+
+async fn post_row(
+    Path((table, row_number)): Path<(String, String)>,
+    State(state): State<Arc<AppState>>,
+    Query(query_params): Query<RequestParams>,
+    Json(form_params): Json<RequestParams>,
```

Comment:
This should be `Form(form_params): Form<RequestParams>`, and then Axum will automatically parse the `application/x-www-form-urlencoded` content of the POSTed form.